src/gleam/list.gleam

```diff
@@ -1361,7 +1369,10 @@ pub fn partition(
 }
 
 /// Returns all the permutations of a list.
-/// All values must be unique.
+///
+/// ## Notice
+///
+/// All list item values must be unique.
```

Comment:
Why is this? What happens otherwise?

Reply:
1. The comment about uniqness of item-values for permutation existed prior.

2. I tried to make it more prominent because in my mind the function could check and return a Result if items are of non-unique values.

3. AFAIU, the reason it does not work is the implementation which checks against values not against index. A permutation by index of `[1, 1, 1]` would still be `3 * 2 * 1 = 6`:

Permutations of `[1, 2, 3]` by value or index:

```gleam
[1, 2, 3]
[1, 3, 2]
[2, 1, 3]
[2, 3, 1]
[3, 1, 2]
[3, 2, 1]
```

Permutations of `[1, 1, 1]` by value:

```gleam
[1, 1, 1]
```

Permutations of `[1, 1, 1]` by index:

```gleam
[1, 1, 1]
[1, 1, 1]
[1, 1, 1]
[1, 1, 1]
[1, 1, 1]
[1, 1, 1]
```

But even if it permutates by values, wouldn't that mean that the algorithm would need to reduce the input list of `[1,1,2,3,3]` to `[1,2,3]` for perumtation purposes?

```gleam
let expected = [
  [1, 2, 3],
  [1, 3, 2],
  [2, 1, 3],
  [2, 3, 1],
  [3, 1, 2],
  [3, 2, 1],
]

[1, 1, 2, 3, 3]
|> list.permutations
|> should.equal(expected)
```

```
  1) gleam@list_test:permutations_test/0
     Failure: ?assertEqual([[1,2,3],[1,3,2],[2,1,3],[2,3,1],[3,1,2],[3,2,1]], Actual)
       expected: [[1,2,3],[1,3,2],[2,1,3],[2,3,1],[3,1,2],[3,2,1]]
            got: [[1,2,3],
                  [1,2,3],
                  [1,3,2],
                  [1,3,2],
                  [1,2,3],
                  [1,2,3],
                  [1,3,2],
                  [1,3,2],
                  [2,1,3],
                  [2,1,3],
                  [2,1,3],
                  [2,1,3],
                  [2,3,1],
                  [2,3,1],
                  [2,3,1],
                  [2,3,1],
                  [3,1,2],
                  [3,1,2],
                  [3,2,1],
                  [3,2,1],
                  [3,1,2],
                  [3,1,2],
                  [3,2,1],
                  [3,2,1]]
```